### PR TITLE
misc: add workflow to roll build files to exports branch

### PR DESCRIPTION
**Pull request by connorjclark**
Googlers, see [this](https://docs.google.com/document/d/1Ll60IwHuej0vD2tDFz6MdY2a4yaimIj1ilZHLhA6MQI/edit#).

### CODE REVIEW

**Comment by brendankenny** [comment]
It seems like workflow artifacts (or can you have a release asset for HEAD?) seem like a better primitive for this, and we're already `yarn build-all`ing there, but really Option 3 in that doc seems like a better choice overall. It's essentially the same thing but done not in a Github action and not pushed to a magic branch.

File: .github/workflows/roll-to-exports-branch.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    runs-on: ubuntu-latest
+    strategy:
+      # e.g. if lint fails, continue to the unit tests anyway
+      fail-fast: false
+
+    steps:
+    - name: git clone
+      uses: actions/checkout@v2
+    - run: git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
```
Comment (paulirish) [inline .github/workflows/roll-to-exports-branch.yml:21]: what's this extra fetch about?

File: .github/workflows/roll-to-exports-branch.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  roll-to-exports-branch:
+
+    runs-on: ubuntu-latest
+    strategy:
+      # e.g. if lint fails, continue to the unit tests anyway
```
Comment (paulirish) [inline .github/workflows/roll-to-exports-branch.yml:15]: remove the fail-fast.

also turns out this only affects matrix jobs and so it's not what we/i commented it as.

File: .github/workflows/roll-to-exports-branch.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      uses: actions/cache@v1
+      id: yarn-cache
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+
+    - run: yarn --frozen-lockfile
+    - run: yarn build-all
```
Comment (paulirish) [inline .github/workflows/roll-to-exports-branch.yml:42]: do we need to build **all**?

File: .github/workflows/roll-to-exports-branch.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    - run: git config --local user.name "Github Action"
+    - run: yarn gh-pages --branch exports --dist exports_output --no-push
+    - run: ls node_modules/gh-pages/.cache
+
+    # Must use this action to authorize correctly w/ GitHub.
+    - name: Push changes
+      uses: ad-m/github-push-action@f18534
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        directory: node_modules/gh-pages/.cache/github.com!GoogleChrome!lighthouse
```
Comment (paulirish) [inline .github/workflows/roll-to-exports-branch.yml:58]: wtf is "Directory to change to before pushing."? https://github.com/ad-m/github-push-action#inputs

File: .github/workflows/roll-to-exports-branch.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+
+    - run: yarn --frozen-lockfile
+    - run: yarn build-all
+    - run: mkdir exports_output
+    - run: mv dist exports_output/dist
+    - run: echo "${GITHUB_SHA::8}" > exports_output/commit_sha
+
+    # gh-pages makes a shallow copy of the repo in cache and adds an unpublished commit.
```
Comment (paulirish) [inline .github/workflows/roll-to-exports-branch.yml:47]:
```suggestion
    # Use gh-pages util to make a standalone repo from the exports_output folder and don't push
```

File: .github/workflows/roll-to-exports-branch.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      id: yarn-cache
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+
+    - run: yarn --frozen-lockfile
+    - run: yarn build-all
+    - run: mkdir exports_output
```
Comment (paulirish) [inline .github/workflows/roll-to-exports-branch.yml:43]:
```suggestion

    - run: mkdir exports_output
```

File: .github/workflows/roll-to-exports-branch.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+# Roll build files to exports branch.
```
Comment (paulirish) [inline .github/workflows/roll-to-exports-branch.yml:1]: bikeshed: feels more like a 'deploy' than a 'roll'

**Comment by patrickhulce** [comment]
we're not doing this anymore

**Pull request closed without merging**